### DISCUSSION TITLE
How to convert a string timestamp to timestamp datatype column?

### DISCUSSION THREAD

**Discussion by rushi-jariwala** [Eventual-Inc/Daft#2294] [Q&A]
I have a timestamp column in string, something like this:
![image](https://github.com/Eventual-Inc/Daft/assets/50650888/d46f43c6-f2a9-4307-bf5b-bd1b703e6c01)

I want to cast this to timestamp column. I am unable to find any documentation on this. 

I tried this, but it did not work. Output is None. 
`df = df.with_column('timestamp1',col('timestamp').cast(DataType.timestamp('us')))`

**Comment by samster25** [comment]
@kevinzwang, can you spec out an expression for converting strings and other types into a TimeStamp. Similar to https://pandas.pydata.org/docs/reference/api/pandas.to_datetime.html

**Reply by kevinzwang** [reply]
Will do! I was just thinking about this recently

**Comment by kevinzwang** [comment]
I'll be tracking progress on this at #2302
